backends/cdc-mongo/relay.go

```diff
@@ -86,25 +55,34 @@ func (r *Relayer) Relay() error {
 		database := r.Service.Database(r.Options.CDCMongo.Database)
 		if r.Options.CDCMongo.Collection == "" {
 			// Watch specific database and all collections under it
-			cs, err = database.Watch(ctx, mongo.Pipeline{}, streamOpts...)
+			cs, err = database.Watch(r.ShutdownContext, mongo.Pipeline{}, streamOpts...)
 		} else {
 			// Watch specific database and collection deployment
 			coll := database.Collection(r.Options.CDCMongo.Collection)
-			cs, err = coll.Watch(ctx, mongo.Pipeline{}, streamOpts...)
+			cs, err = coll.Watch(r.ShutdownContext, mongo.Pipeline{}, streamOpts...)
 		}
 	} else {
 		// Watch entire deployment
-		cs, err = r.Service.Watch(ctx, mongo.Pipeline{}, streamOpts...)
+		cs, err = r.Service.Watch(r.ShutdownContext, mongo.Pipeline{}, streamOpts...)
 	}
 
 	if err != nil {
 		return errors.Wrap(err, "could not begin change stream")
 	}
 
-	defer cs.Close(ctx)
+	defer cs.Close(r.ShutdownContext)
 
 	for {
-		if !cs.Next(ctx) {
+		select {
+		case <-r.ShutdownContext.Done():
+			r.log.Info("Received shutdown signal, existing relayer")
+			return nil
+		default:
+			// noop
+		}
+
+		// TODO: test this. Next blocks, but does the underlaying code handle the context correctly?
+		if !cs.Next(r.ShutdownContext) {
 			r.log.Errorf("unable to read message from mongo: %s", cs.Err())
```

Comment:
This is an oversight - it needs a sleep + continue.

Reply:
Fixed!